crates/bevy_math/src/linear_convex_space.rs

```diff
@@ -0,0 +1,69 @@
+//! This module provides the `LinearConvexSpace` trait and some related objects
+use glam::{DVec2, DVec3, DVec4, Quat, UVec2, UVec3, UVec4, Vec2, Vec3, Vec3A, Vec4};
+
+/// A trait providing methods for spanning a linear, convex space.
+///
+/// A convex space or convex set is a set of points (think points in 3d or 2d space) in which every point on a line between two points A and B in that set
+/// is also a part of that set. For example the set of all points inside a square is a convex set as you can draw a line between any two points in the square and never leave it.
+/// For more information, please see [this wikipedia article](https://en.wikipedia.org/wiki/Convex_set).
+///
+/// The space should also be linear, meaning that `A*t + B*(1 - t)` with any points `A`, `B` and `t` element `0..1` should represent a straight line.
+///
+/// By implementing this trait, you guarantee that the above conditions hold true.
+pub trait LinearConvexSpace<Scalar>: Default + Copy + Clone {
```

Comment:
I would put the `Scalar` generic as an associated type instead, as the notion of "scalar type" is dependent on the type of the vector, and not something is parametrized.
```suggestion
pub trait LinearConvexSpace: Default + Copy + Clone {
    type Scalar: Copy + Add + Sub + Mul + Div /* and maybe extra bounds */;
```

This way you can make the implementations using this trait fully generic easily, without the semi-arbitrary restriction on `f32` as you have done for the cubic splines implementation above.

Reply:
That's fine by me!